prow/config/config.go

```diff
@@ -767,6 +798,10 @@ func loadConfig(prowConfig, jobConfig string) (*Config, error) {
 
 	nc.ProwYAMLGetter = defaultProwYAMLGetter
 
+	if nc.InRepoConfig.AllowedClusters == nil {
```

Comment:
We should set the `*` key if it is unset, not only if the entire map is unspecified. Please add a test case to validate this.